src/libraries/System.Runtime/tests/System/Reflection/RuntimeTypeMemberCacheTests.cs

```diff
@@ -0,0 +1,368 @@
+            MemberInfo[] methods = t.GetMember("Method3");
+            Assert.NotNull(method3Int);
+            Assert.NotNull(method3String);
+            Assert.Equal(0, ((MethodInfo)methods[0]).GetParameters().Length);
+            Assert.Equal(method3String, methods[1]);
+            Assert.Equal(method3Int, methods[2]);
+        }
+
+        [Fact]
+        public static void TypeGetConstructorsReturnsInDeclaredOrderTest()
+        {
+            Type t = typeof(Test);
+            ConstructorInfo constructor3 = t.GetConstructor(new Type[] { typeof(string) });
+            Assert.NotNull(constructor3);
+            ConstructorInfo[] constuctors = t.GetConstructors();
+            Assert.Equal(typeof(int), constuctors[0].GetParameters()[0].ParameterType);
+            Assert.Equal(0, constuctors[1].GetParameters().Length);
+            Assert.Equal(constructor3, constuctors[2]);
+        }
+
+        [Fact]
+        public static void TypeGetEventsReturnsInDeclaredOrderTest()
+        {
+            Type t = typeof(Test);
+            EventInfo event2 = t.GetEvent("Event2");
+            Assert.NotNull(event2);
+            EventInfo[] events = t.GetEvents();
+            Assert.Equal("Event1", events[0].Name);
+            Assert.Equal(event2, events[1]);
+            Assert.Equal("Event3", events[2].Name);
+        }
+
+        [Fact]
+        public static void TypeGetPropertiesReturnsInDeclaredOrderTest()
+        {
+            Type t = typeof(Test);
+            PropertyInfo propertyBase = t.GetProperty("PBase");
+            PropertyInfo property3 = t.GetProperty("Property3");
+            Assert.NotNull(propertyBase);
+            Assert.NotNull(property3);
+            PropertyInfo[] properties = t.GetProperties();
+            Assert.Equal("Property1", properties[0].Name);
+            Assert.Equal("Property2", properties[1].Name);
+            Assert.Equal(property3, properties[2]);
+            Assert.Equal(propertyBase, properties[4]);
+        }
+
+        [Fact]
+        public static void TypeGetFieldsReturnsInDeclaredOrderTest()
+        {
+            Type t = typeof(Test);
+            FieldInfo field3 = t.GetField("field3");
+            Assert.NotNull(field3);
+            FieldInfo[] fields = t.GetFields();
+            Assert.Equal("field1", fields[0].Name);
+            Assert.Equal("field2", fields[1].Name);
+            Assert.Equal(field3, fields[2]);
+        }
+
+        [Fact]
+        public static void BigTypeGetMethodsReturnsInDeclaredOrderTest()
+        {
+            Type t = typeof(BigTestType);
+            MethodInfo method129 = t.GetMethod("Method129");
+            MethodInfo method57 = t.GetMethod("Method57");
+            FieldInfo field = t.GetField("field2");
+            Assert.NotNull(method129);
+            Assert.NotNull(method57);
+
+            for (int i = 150; i > 100; i--)
+            {
+                MethodInfo m = t.GetMethod($"Method{i}");
```

Comment:
Is calling `GetMethod()` sufficient here? If this is being used for its side-effect of priming the cache, I would add a comment for its purpose so it isn't removed by a future person.